powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/entity/ActivationHistoryEntity.java

```diff
@@ -216,54 +192,32 @@ public void setActivationVersion(Integer version) {
         this.activationVersion = version;
     }
 
+    public String getActivationName() {
+        return activationName;
+    }
+
+    public void setActivationName(final String activationName) {
+        this.activationName = activationName;
+    }
+
     @Override
     public int hashCode() {
-        int hash = 7;
-        hash = 23 * hash + Objects.hashCode(this.activation);
-        hash = 23 * hash + Objects.hashCode(this.activationStatus);
-        hash = 23 * hash + Objects.hashCode(this.eventReason);
-        hash = 23 * hash + Objects.hashCode(this.externalUserId);
-        hash = 23 * hash + Objects.hashCode(this.timestampCreated);
-        hash = 23 * hash + Objects.hashCode(this.activationVersion);
-        return hash;
+        // hashCode should always return the same value even when autogenerated ID has been set; performance penalty is negligible
+        return getClass().hashCode();
```

Comment:
Let's not add the new field `applicationName` and return constant as suggested https://vladmihalcea.com/how-to-implement-equals-and-hashcode-using-the-jpa-entity-identifier/
If agreed, I will update our guidelines.